EXILED/Exiled.API/Features/Items/MicroHid.cs

```diff
@@ -49,23 +81,115 @@ public float Energy
         public new MicroHIDItem Base { get; }
 
         /// <summary>
-        /// Gets or sets the <see cref="HidState"/>.
+        /// Gets or sets a value indicating whether the MicroHID is broken.
+        /// </summary>
+        public bool IsBroken
+        {
+            get => BrokenModule.Broken;
+            set => BrokenModule.ServerSetBroken(Serial, value);
+        }
+
+        /// <summary>
+        /// Gets a time when this <see cref="MicroHid"/> was broken.
+        /// </summary>
+        /// <value>A time when this <see cref="MicroHid"/> was broken, or <c>0</c> if it is not broken.</value>
+        public float BrokeTime => BrokenSyncModule.TryGetBrokenElapsed(Serial, out float time) ? time : 0;
+
+        /// <summary>
+        /// Gets or sets the <see cref="MicroHidPhase"/>.
+        /// </summary>
+        public MicroHidPhase State
+        {
+            get => CycleController.Phase;
+            set => CycleController.Phase = value;
+        }
+
+        /// <summary>
+        /// Gets or sets progress of winging up.
+        /// </summary>
+        /// <value>A value between <c>0</c> and <c>1</c>.</value>
+        public float WindUpProgress
+        {
+            get => CycleController.ServerWindUpProgress;
+            set => CycleController.ServerWindUpProgress = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the last received <see cref="MicroHidFiringMode"/>.
+        /// </summary>
+        public MicroHidFiringMode LastFiringMode
+        {
+            get => CycleController.LastFiringMode;
+            set => CycleController.LastFiringMode = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the last received <see cref="InputSyncModule.SyncData"/>.
         /// </summary>
-        public HidState State
+        public InputSyncModule.SyncData LastReceived
         {
-            get => Base.State;
-            set => Base.State = value;
+            get => InputModule._lastReceived;
+            set => InputModule._lastReceived = value;
         }
 
+        /// <summary>
+        /// Gets a value indicating whether the <see cref="LastReceived"/> is <see cref="InputSyncModule.SyncData.Primary"/>.
+        /// </summary>
+        public bool IsPrimary => InputModule.Primary;
+
         /// <summary>
         /// Starts firing the MicroHID.
         /// </summary>
-        public void Fire() => Base.Fire();
+        /// <param name="firingMode">Fire mode.</param>
+        public void Fire(MicroHidFiringMode firingMode = MicroHidFiringMode.PrimaryFire)
+        {
+            if (TryGetFireController(firingMode, out PrimaryFireModeModule module))
+                module.GetType().GetMethod("ServerFire", BindingFlags.Public | BindingFlags.DeclaredOnly)!.Invoke(module, Array.Empty<object>());
```

Comment:
Save it

Reply:
MethodInfo?

Reply:
yes

Reply:
shouldn't this method call this instead ServerUpdateSelected(MicroHidPhase MicroHidPhase.Firing) from FiringModeControllerModule